akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -72,9 +72,24 @@ sealed trait HttpEntity extends jm.HttpEntity {
    */
   def toStrict(timeout: FiniteDuration)(implicit fm: Materializer): Future[HttpEntity.Strict] =
     dataBytes
-      .via(new akka.http.impl.util.ToStrict(timeout, contentType))
+      .via(new akka.http.impl.util.ToStrict(timeout, maxBytes = None, contentType))
       .runWith(Sink.head)
 
+  /**
+   * Collects all possible parts and returns a potentially future Strict entity for easier processing.
+   * The Future is failed with an TimeoutException if the stream isn't completed after the given timeout,
+   * or with a FooException when the end of the entity is not reached within the maximum number of bytes.
+   */
+  def toStrict(timeout: FiniteDuration, maxBytes: Long)(implicit fm: Materializer): Future[HttpEntity.Strict] = contentLengthOption match {
+    case Some(contentLength) if contentLength > maxBytes ⇒
+      FastFuture.failed(new EntityStreamException(new ErrorInfo("Request too large", s"Request of size $contentLength was longer than the maximum of $maxBytes")))
+    case _ ⇒
+      // TODO can we trust the content-length or should we check the actual length anyway?
```

Comment:
WDYM? `ToStrict` will check the actual size later, doesn't it?

Reply:
Yes, right now I chose the 'safe' path. I wondered whether there could be cases where we could decide early that it's not needed to add the `ToStrict`. I guess the only case is `Strict`, which might not be worth special-casing. I'll remove the comment ;)